agent.go

```diff
@@ -622,44 +635,83 @@ type initMount struct {
-	{"cgroup", "cgroup", "/sys/fs/cgroup/cpu,cpuacct", []string{"nosuid", "nodev", "noexec", "relatime", "cpuacct", "cpu"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/pids", []string{"nosuid", "nodev", "noexec", "relatime", "pids"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/net_cls,net_prio", []string{"nosuid", "nodev", "noexec", "relatime", "net_prio", "net_cls"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/blkio", []string{"nosuid", "nodev", "noexec", "relatime", "blkio"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/freezer", []string{"nosuid", "nodev", "noexec", "relatime", "freezer"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/cpuset", []string{"nosuid", "nodev", "noexec", "relatime", "cpuset"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/memory", []string{"nosuid", "nodev", "noexec", "relatime", "memory"}},
-	{"cgroup", "cgroup", "/sys/fs/cgroup/perf_event", []string{"nosuid", "nodev", "noexec", "relatime", "perf_event"}},
-	//{"cgroup", "cgroup", "/sys/fs/cgroup/hugetlb", []string{"nosuid", "nodev", "noexec", "relatime", "hugetlb"}},
-	{"bind", "/sys/fs/cgroup/cpu,cpuacct", "/sys/fs/cgroup/cpu", []string{"bind"}},
-	{"bind", "/sys/fs/cgroup/cpu,cpuacct", "/sys/fs/cgroup/cpuacct", []string{"bind"}},
-	{"bind", "/sys/fs/cgroup/net_cls,net_prio", "/sys/fs/cgroup/net_cls", []string{"bind"}},
-	{"bind", "/sys/fs/cgroup/net_cls,net_prio", "/sys/fs/cgroup/net_prio", []string{"bind"}},
-	{"tmpfs", "tmpfs", "/sys/fs/cgroup", []string{"remount", "ro", "nosuid", "nodev", "noexec", "mode=755"}},
+func getCgroupMounts(cgPath string) ([]initMount, error) {
+	f, err := os.Open(cgPath)
+	if err != nil {
+		return []initMount{}, err
+	}
+	defer f.Close()
+
+	cgroupMounts := []initMount{{"tmpfs", "tmpfs", cgroupPath, []string{"nosuid", "nodev", "noexec", "mode=755"}}}
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		text := scanner.Text()
+		fields := strings.Split(text, "\t")
+		cgroup := fields[0]
+
+		if cgroup == "" || cgroup[0] == '#' || (len(fields) > 3 && fields[3] == "0") {
+			continue
+		}
+		cgroupMounts = append(cgroupMounts, initMount{"cgroup", "cgroup",
+			filepath.Join(cgroupPath, cgroup), []string{"nosuid", "nodev", "noexec", "relatime", cgroup}})
+	}
+
+	if err = scanner.Err(); err != nil {
+		return []initMount{}, err
+	}
+
+	cgroupMounts = append(cgroupMounts, initMount{"tmpfs", "tmpfs",
+		cgroupPath, []string{"remount", "ro", "nosuid", "nodev", "noexec", "mode=755"}})
+	return cgroupMounts, nil
 }
 
-// initAgentAsInit will do the initializations such as setting up the rootfs
-// when this agent has been run as the init process.
-func initAgentAsInit() error {
+func mountToRootfs(m initMount) error {
+	if err := os.MkdirAll(m.dest, os.FileMode(0755)); err != nil {
+		return err
+	}
+
+	if flags, options, err := parseMountFlagsAndOptions(m.options); err != nil {
+		return grpcStatus.Errorf(codes.Internal, "Could parseMountFlagsAndOptions(%v)", m.options)
```

Comment:
s/Could parseMountFlagsAndOptions(%v)/Could not parse mount flags %v/

Reply:
oh, sorry, I missed out it , thanks.